python/python/tests/test_search.py

```diff
@@ -0,0 +1,176 @@
+# Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+
+from typing import List
+
+import pytest
+from glide.async_commands.core import InfoSection
+from glide.async_commands.redis_modules.search.commands import Search
+from glide.async_commands.redis_modules.search.field import (
+    CoordinateSystem,
+    Field,
+    GeoField,
+    GeoShapeField,
+    NumericField,
+    TagField,
+    TextField,
+    VectorAlgorithm,
+    VectorField,
+)
+from glide.async_commands.redis_modules.search.index import Index, IndexType
+from glide.constants import OK
+from glide.exceptions import RequestError
+from glide.redis_client import TRedisClient
+from tests.test_async_client import (
+    check_if_server_version_lt,
+    get_random_string,
+    parse_info_response,
+)
+
+
+@pytest.mark.asyncio
+class TestSearchModule:
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    async def test_search_module_loaded(self, redis_client: TRedisClient):
+        res = parse_info_response(await redis_client.info([InfoSection.MODULES]))
+        assert "search" in res["module"]
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    async def test_json_module_loadded(self, redis_client: TRedisClient):
+        res = parse_info_response(await redis_client.info([InfoSection.MODULES]))
+        assert "ReJSON" in res["module"]
+
+    @pytest.mark.parametrize("cluster_mode", [True, False])
+    async def test_create_index_hash(self, redis_client: TRedisClient):
+        search = Search()
+        idx_name = get_random_string(5)
+        index_definition = Index(
+            name=idx_name, filter="@age>15", prefix=["man:", "woman:"]
+        )
+
+        schema = [
+            TextField(name="first_name"),
+            TextField(name="last_name"),
+            NumericField(name="age"),
+            GeoField(name="location"),
+        ]
+
+        assert (
```

Comment:
all of these tests only test that the call returns OK. Is this really the most meaningful test available?

Reply:
for now, when I add FT.SEARCH the tests will be more meaningful

Reply:
please also add tests for edge cases - passing an empty string for prefix/filter, combining mutually exclusive args, passing nulls etc.. make sure we raise appropriate exceptions